### PR TITLE
Userland: Add an `image` utility

### PR DESCRIPTION
**Pull request by nico**
At the moment, all it can do is read all image formats that LibGfx can read and save to any image format that LibGfx can write (currently bmp, png, qoi).

Currently, it drops all image metadata (including color profiles).

Over time, this could learn tricks like keeping color profiles, converting an image to a different color profile, cropping out a part of an image, and so on.

---

We pay the usual 10x serenity tax atm:

```
% time sips -s format png ~/Downloads/PXL_20220305_152555974.jpg -o tmp.png
/Users/thakis/Downloads/PXL_20220305_152555974.jpg
  /Users/thakis/src/serenity/tmp.png
sips -s format png ~/Downloads/PXL_20220305_152555974.jpg -o tmp.png  0.71s user 0.05s system 96% cpu 0.789 total
% time Build/lagom/image ~/Downloads/PXL_20220305_152555974.jpg -o tmp.png 
Build/lagom/image ~/Downloads/PXL_20220305_152555974.jpg -o tmp.png  9.13s user 0.11s system 99% cpu 9.255 total
```

### CODE REVIEW

File: Userland/Utilities/image.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+ErrorOr<int> serenity_main(Main::Arguments arguments)
+{
+    Core::ArgsParser args_parser;
+
+    StringView in_path;
+    args_parser.add_positional_argument(in_path, "Path to input image file", "FILE");
+
+    StringView out_path;
+    args_parser.add_option(out_path, "Path to output image file", "output", 'o', "FILE");
```
Comment (linusg) [inline Userland/Utilities/image.cpp:23]: having one as a positional argument and the other as an option seems somewhat odd, why not `image <input> <output>`?

File: Userland/Utilities/image.cpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    // over selecting a frame, access color profile data, and so on in the future.
+    auto frame = TRY(decoder->frame(0)).image;
+
+    ByteBuffer bytes;
+    if (out_path.ends_with(".bmp"sv, CaseSensitivity::CaseInsensitive)) {
+        bytes = Gfx::BMPWriter().dump(frame);
+    } else if (out_path.ends_with(".png"sv, CaseSensitivity::CaseInsensitive)) {
+        bytes = TRY(Gfx::PNGWriter::encode(*frame));
+    } else if (out_path.ends_with(".qoi"sv, CaseSensitivity::CaseInsensitive)) {
+        bytes = Gfx::QOIWriter::encode(*frame);
```
Comment (linusg) [inline Userland/Utilities/image.cpp:45]: looks like we went for unique API per writer :yakkie:

**Comment by nico** [comment]
This might go better after #17818.

File: Userland/Utilities/image.cpp (hunk 2 above)
Comment (nico) [reply Userland/Utilities/image.cpp:45]: Yes, I'll unify this a bit later.

File: Userland/Utilities/image.cpp (hunk 1 above)
Comment (nico) [reply Userland/Utilities/image.cpp:23]: Using `-o` for output seems pretty standard. For `<input> <output>` it's never quite clear which one is which too.

Comment (nico) [reply Userland/Utilities/image.cpp:23]: And you could imagine things like `-o .png` or `--to-format=png` or similar which writes to the same filename as the input with just the extension replaced with ".png". That's weirder to do if it's `<input> <output>`.

**Pull request merged**